### ISSUE TITLE
Update to support parsing and providing `stream` attribute

### ISSUE THREAD

**Issue by ronaldtse** [relaton/relaton-ietf#116]
```xml
        <stream>IAB</stream>
        <stream>IETF</stream>
        <stream>INDEPENDENT</stream>
        <stream>IRTF</stream>
        <stream>Legacy</stream>
```

https://www.rfc-editor.org/rfc-index.xml now provides the `<stream>` attribute that accepts the following values:

* IAB
* IETF
* Independent
* IRTF
* Legacy
* Editorial (see #115)

**Comment by andrew2net** [comment]
@ronaldtse implemented in the relaton-ietf v 1.14.3 https://github.com/relaton/relaton-data-rfcs/blob/e34bf36d3e73e42ea84b6bf7ccbe5e1397493f93/data/RFC0001.yaml#L68

**Comment by ronaldtse** [comment]
@andrew2net there are 2 issues:

1. The `stream` attribute does not seem to be validated with the values listed above.
2. The Relaton model/YAML output does not provide `stream`. The `stream` attribute is needed for correctly citing IETF documents.

A follow up to https://github.com/relaton/relaton-py/issues/62 may be necessary.

**Comment by andrew2net** [comment]
@ronaldtse I found in a related issue the comment https://github.com/ietf-tools/bibxml-service/issues/301#issuecomment-1450001700
It seems we shouldn't use `series` to store the `stream`. So we need to add the `stream` to grammar. Ping @opoudjis.

**Comment by ronaldtse** [comment]
The stream is indeed not related to the series. Please help update accordingly. Thanks.

**Comment by opoudjis** [comment]
> It seems we shouldn't use series to store the stream. So we need to add the stream to grammar. Ping @opoudjis.

Where are either of you getting that conclusion from? I'm not seeing that in the linked ticket.

Yes, I can change the grammar to create /ext/stream instead of using series. 

But I am not going to change the grammar of anything without an explicit reason. And nobody, not Andrej, not Ronald, and certainly not R J Sparks, is providing a reason here. 

There is also a reason why stream has been encoded where it is, which is that that is where RFC XML expects to see it, and any change in grammar is going to force me to recode metanorma-ietf.

If I am going to spend an added hour I do not have on changing the position of stream, somebody is going to have to give me a reason to.

**Comment by andrew2net** [comment]
@ronaldtse what is the decision? Can the current implementation be accepted? Or do we need to add the `stream` element?

**Comment by ronaldtse** [comment]
> But I am not going to change the grammar of anything without an explicit reason. And nobody, not Andrej, not Ronald, and certainly not R J Sparks, is providing a reason here.

The reason is that the Series is: RFC, Internet-Draft. Each document in each of these series can belong to a different stream. Hence stream is an orthogonal attribute to series. 

Furthermore, the stream attribute determines how the document is being cited. 

@opoudjis please help facilitate the change.

**Comment by ronaldtse** [comment]
> @ronaldtse what is the decision? Can the current implementation be accepted? Or do we need to add the `stream` element?

We need to add the stream element.

**Comment by opoudjis** [comment]
Since `stream` is an IETF-specific notion, it is only going to appear in the extension point for IETF. 

Very strongly request that you retain the redundancy of both two series types, and the new ext/stream element, encoding the same information. There are information modelling reasons (which I will not reiterate here) why the current design was chosen.

**Comment by opoudjis** [comment]
I've added the `ext/stream`, with enum value.

To be explicit, I request that _IETF series_ and _IETF stream_ be encoded as following:

_IETF series_ : `/series[@type = 'main']/title`, `/series[@type = 'main']/number`
_IETF stream_: `/series[@type = 'stream']/title`, `/ext/stream`

So now we have:

````
  <series type="stream">
    <title format="text/plain">IETF</title>
  </series>
  <series>
    <title format="text/plain">RFC</title>
    <number>7771</number>
  </series>
````

I am proposing this becomes

````
  <series type="main">
    <title format="text/plain">RFC</title>
    <number>7771</number>
  </series>
  <series type="stream">
    <title format="text/plain">IETF</title>
  </series>
...
  <ext>
    ....
    <stream>IETF</stream>
    ....
  </ext>
````

The stream ends up encoded twice, but that helps keep the citation model simple (it does not need to know about flavour-specific extensions.)

**Comment by andrew2net** [comment]
fixed in v 1.14.5
```
relaton fetch 'RFC 8341'
[relaton-ietf] ("RFC 8341") fetching...
[relaton-ietf] ("RFC 8341") found RFC 8341
<bibdata type="standard" schema-version="v1.2.3">
  ...
  <series>
    <title format="text/plain">STD</title>
    <number>91</number>
  </series>
  <series>
    <title format="text/plain">RFC</title>
    <number>8341</number>
  </series>
  <series type="stream">
    <title format="text/plain">IETF</title>
  </series>
  ...
  <ext schema-version="v1.0.0">
    <editorialgroup>
      <committee>netconf</committee>
    </editorialgroup>
    <stream>IETF</stream>
  </ext>
</bibdata>
```
